examples/simple_main.cc

```diff
@@ -0,0 +1,10 @@
+#include "simple_common.h"
+
+int main(void)
+{
+    CustomData data = {};
+    memset(&data, 0, sizeof(data));
+
+    Engine catalyst;
+    return catalyst.run("examples/simple.cc", (void *)&data, sizeof(data));
```

Comment:
A: It's what I do to support static compilation without any overhead for function pointers. I don't think it's an uncommon approach, but it may look weird, yes.
B: I guess I could use my own string concatenation function that doesn't keep iterating over the string.
C: We may want to chat about what I did so we can come up with something better for both Mac and Linux.
D: Ah, that's true. I assumed a "unity build" I suppose. In that case, we should provide some way to recompile the game code on the programmer's own terms. (For example, a path to a build script file instead.) The dylib check is something along the lines of what I do.